lib/python/ray/worker.py

```diff
@@ -158,11 +159,28 @@ def put_object(self, objref, value):
       objref (ray.ObjRef): The object reference of the value to be put.
       value (serializable object): The value to put in the object store.
     """
-    if serialization.is_arrow_serializable(value):
-      ray.lib.put_arrow(self.handle, objref, value)
-    else:
-      object_capsule, contained_objrefs = serialization.serialize(self.handle, value) # contained_objrefs is a list of the objrefs contained in object_capsule
-      ray.lib.put_object(self.handle, objref, object_capsule, contained_objrefs)
+
+    try:
+      serialized = libnumbuf.serialize_list([value])
+      schema = libnumbuf.get_schema_metadata(serialized)
+      # TODO(pcm): Right now, metadata is serialized twice, change that in the future
+      # in the following line, the "8" is for storing the metadata size,
+      # the len(schema) is for storing the metadata and the 4096 is for storing
+      # the metadata in the batch (see INITIAL_METADATA_SIZE in arrow)
+      size = libnumbuf.get_serialized_size(serialized) + 8 + len(schema) + 4096
+      buff, segmentid = ray.lib.allocate_buffer(self.handle, objref, size)
+      # write the metadata length
+      np.frombuffer(buff, dtype="int64", count=1)[0] = len(schema)
+      # metadata buffer
+      metadata = np.frombuffer(buff, dtype="byte", offset=8, count=len(schema))
+      # write the metadata
+      metadata[:] = schema
+      data = np.frombuffer(buff, dtype="byte", offset=8+len(schema))
+      metadata_offset = libnumbuf.write_to_buffer(serialized, memoryview(data))
+      ray.lib.finish_buffer(self.handle, objref, segmentid, metadata_offset)
+    except:
+        object_capsule, contained_objrefs = serialization.serialize(self.handle, value) # contained_objrefs is a list of the objrefs contained in object_capsule
```

Comment:
is there a way to check that this path is only used when we think it is being used?